src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +34,15 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        double oddAverage = IntStream.range(0, numbers.size())
+                .mapToObj(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(num -> num % 2 == 1)
+                .mapToDouble(Integer::doubleValue)
```

Comment:
Reduce `map` using in the stream as much as possible. 

```suggestion
                .mapToDouble(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
                .filter(num -> num % 2 == 1)
```